// client/src/pages/Achievements.jsx
// client/src/pages/Achievements.jsx
import React, { useEffect, useMemo, useState } from 'react';
import { motion } from 'framer-motion';
import { pageTransition, staggerContainer } from '../utils/animations';
import ScrollReveal from '../components/animations/ScrollReveal';
import Card from '../components/common/Card';
import Badge from '../components/common/Badge';
import SearchBar from '../components/common/SearchBar';
import FilterDropdown from '../components/common/FilterDropdown';
import Modal from '../components/common/Modal';
import api from '../utils/api';

const Achievements = () => {
  const [search, setSearch] = useState('');
  const [category, setCategory] = useState('');
  const [level, setLevel] = useState('');
  const [year, setYear] = useState('');
  const [achievements, setAchievements] = useState([]);
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState('');
  const [selectedAchievement, setSelectedAchievement] = useState(null);

  useEffect(() => {
    const fetchAchievements = async () => {
      try {
        setLoading(true);
        setError('');
        const res = await api.get('/achievements');
        setAchievements(Array.isArray(res.data) ? res.data : []);
      } catch (err) {
        console.error('Gagal mengambil data prestasi:', err);
        setError('Gagal memuat data dari server.');
        setAchievements([]);
      } finally {
        setLoading(false);
      }
    };

    fetchAchievements();
  }, []);

  const categories = useMemo(
    () => Array.from(new Set(achievements.map((a) => a.category))).filter(Boolean),
    [achievements]
  );

  const levels = useMemo(
    () => Array.from(new Set(achievements.map((a) => a.level))).filter(Boolean),
    [achievements]
  );

  const years = useMemo(
    () =>
      Array.from(new Set(achievements.map((a) => a.year)))
        .filter(Boolean)
        .sort((a, b) => b - a),
    [achievements]
  );

  const filtered = useMemo(() => {
    return achievements.filter((a) => {
      const matchSearch =
        a.title.toLowerCase().includes(search.toLowerCase()) ||
        a.studentName.toLowerCase().includes(search.toLowerCase());
      const matchCategory = category ? a.category === category : true;
      const matchLevel = level ? a.level === level : true;
      const matchYear = year ? String(a.year) === String(year) : true;
      return matchSearch && matchCategory && matchLevel && matchYear;
    });
  }, [achievements, search, category, level, year]);

  const totalByYear = (targetYear) =>
    achievements.filter((a) => a.year === targetYear).length;

  const handleOpenModal = (item) => setSelectedAchievement(item);
  const handleCloseModal = () => setSelectedAchievement(null);

  return (
    <motion.div
      variants={pageTransition}
      initial="initial"
      animate="animate"
      exit="exit"
      className="space-y-8"
    >
      <section>
        <div className="mb-4 flex flex-col gap-3 lg:flex-row lg:items-end lg:justify-between">
          <div>
            <h1 className="text-xl font-semibold text-text">
              Prestasi Siswa XI RPL
            </h1>
            <p className="text-xs text-text-muted max-w-xl">
              Rekap prestasi siswa XI RPL di berbagai bidang. Data diambil
              langsung dari database dan bisa difilter berdasarkan kategori,
              tingkat, dan tahun.
            </p>
          </div>

          <div className="flex flex-col gap-2 sm:flex-row sm:flex-wrap sm:items-center sm:gap-3">
            <SearchBar
              value={search}
              onChange={setSearch}
              placeholder="Cari judul atau nama siswa..."
              className="w-full sm:w-56"
            />
            <FilterDropdown
              label="Kategori"
              value={category}
              onChange={setCategory}
              options={categories.map((c) => ({ value: c, label: c }))}
            />
            <FilterDropdown
              label="Tingkat"
              value={level}
              onChange={setLevel}
              options={levels.map((l) => ({ value: l, label: l }))}
            />
            <FilterDropdown
              label="Tahun"
              value={year}
              onChange={setYear}
              options={years.map((y) => ({ value: y, label: y }))}
            />
          </div>
        </div>

        {error && !loading && (
          <div className="mb-3 text-xs text-state-warning">
            {error}
          </div>
        )}

        <ScrollReveal>
          {loading ? (
            <div className="grid gap-4 sm:grid-cols-2 xl:grid-cols-3 mb-4">
              {Array.from({ length: 6 }).map((_, idx) => (
                <div key={idx} className="skeleton h-32" />
              ))}
            </div>
          ) : achievements.length === 0 && !error ? (
            <div className="glass-card border border-white/5 p-4 text-xs text-text-muted">
              Belum ada data prestasi di database. Tambahkan prestasi melalui
              panel admin.
            </div>
          ) : (
            <>
              <div className="mb-3 flex flex-wrap items-center justify-between gap-2 text-xs text-text-muted">
                <p>
                  Menampilkan{' '}
                  <span className="font-semibold text-text">
                    {filtered.length}
                  </span>{' '}
                  dari{' '}
                  <span className="font-semibold text-text">
                    {achievements.length}
                  </span>{' '}
                  prestasi.
                </p>
                {years.length > 0 && (
                  <p className="text-[11px]">
                    Tahun terbaru{' '}
                    <span className="font-semibold text-text">{years[0]}</span>{' '}
                    memiliki{' '}
                    <span className="font-semibold text-text">
                      {totalByYear(years[0])}
                    </span>{' '}
                    prestasi.
                  </p>
                )}
              </div>

              {filtered.length === 0 ? (
                <div className="glass-card border border-white/5 p-4 text-xs text-text-muted">
                  Tidak ada prestasi yang cocok dengan filter saat ini.
                </div>
              ) : (
                <motion.div
                  variants={staggerContainer(0.06, 0)}
                  initial="hidden"
                  animate="visible"
                  className="grid gap-4 sm:grid-cols-2 xl:grid-cols-3"
                >
                  {filtered.map((item, index) => (
                    <motion.div
                      key={item.id}
                      variants={{
                        hidden: { opacity: 0, y: 16 },
                        visible: {
                          opacity: 1,
                          y: 0,
                          transition: {
                            duration: 0.35,
                            delay: index * 0.04
                          }
                        }
                      }}
                    >
                      <Card
                        className="flex flex-col gap-3 p-3 cursor-pointer"
                        onClick={() => handleOpenModal(item)}
                      >
                        <div className="flex items-start gap-3">
                          <div className="relative h-10 w-10 overflow-hidden rounded-full bg-card-soft">
                            <img
                              src={item.photo || item.photo_url}
                              alt={item.studentName}
                              className="h-full w-full object-cover"
                              loading="lazy"
                              onError={(e) => {
                                e.currentTarget.style.display = 'none';
                              }}
                            />
                            <div className="absolute inset-0 flex items-center justify-center text-[10px] text-text-muted">
                              {item.studentName
                                .split(' ')
                                .slice(0, 2)
                                .map((n) => n[0])
                                .join('')
                                .toUpperCase()}
                            </div>
                          </div>
                          <div className="flex-1">
                            <p className="text-sm font-semibold text-text line-clamp-2">
                              {item.title}
                            </p>
                            <p className="text-[11px] text-text-muted">
                              {item.studentName}
                            </p>
                          </div>
                        </div>

                        <div className="flex flex-wrap gap-2 text-[10px]">
                          <Badge variant="info">{item.category}</Badge>
                          <Badge variant="purple">{item.level}</Badge>
                          <Badge variant="success">{item.year}</Badge>
                        </div>

                        <p className="text-[11px] text-text-muted line-clamp-3">
                          {item.description}
                        </p>

                        <p className="mt-1 text-[10px] text-primary/80">
                          Klik untuk lihat detail lengkap
                        </p>
                      </Card>
                    </motion.div>
                  ))}
                </motion.div>
              )}
            </>
          )}
        </ScrollReveal>

        <Modal
          isOpen={!!selectedAchievement}
          onClose={handleCloseModal}
          title={selectedAchievement?.title || 'Detail Prestasi'}
        >
          {selectedAchievement && (
            <div className="space-y-3">
              <div className="flex items-center gap-3">
                <div className="relative h-10 w-10 overflow-hidden rounded-full bg-card-soft">
                  <img
                    src={selectedAchievement.photo || selectedAchievement.photo_url}
                    alt={selectedAchievement.studentName}
                    className="h-full w-full object-cover"
                    loading="lazy"
                    onError={(e) => {
                      e.currentTarget.style.display = 'none';
                    }}
                  />
                  <div className="absolute inset-0 flex items-center justify-center text-[10px] text-text-muted">
                    {selectedAchievement.studentName
                      .split(' ')
                      .slice(0, 2)
                      .map((n) => n[0])
                      .join('')
                      .toUpperCase()}
                  </div>
                </div>
                <div>
                  <p className="text-sm font-semibold text-text">
                    {selectedAchievement.studentName}
                  </p>
                  <div className="mt-1 flex flex-wrap gap-1.5 text-[10px]">
                    <Badge variant="info">{selectedAchievement.category}</Badge>
                    <Badge variant="purple">{selectedAchievement.level}</Badge>
                    <Badge variant="success">{selectedAchievement.year}</Badge>
                  </div>
                </div>
              </div>

              <p className="text-[13px] text-text-muted">
                {selectedAchievement.description}
              </p>
            </div>
          )}
        </Modal>
      </section>
    </motion.div>
  );
};

export default Achievements;
